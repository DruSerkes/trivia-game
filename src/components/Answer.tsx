import * as React from 'react';
import { ListItem } from '@material-ui/core'
import { showIfCorrect } from '../helpers/addClass'

type AnswerProps = {
    answer: string;
    answerQuestion: (answer: string) => void;
    correct?: boolean | null;
}

export function Answer({ answer, answerQuestion, correct }: AnswerProps) {
    const renderHTML = () => (<span className="AnswerText" dangerouslySetInnerHTML={{ __html: answer }} />)
    const handleClick = ({ nativeEvent: e }: React.MouseEvent) => {
        showIfCorrect(e, correct);
        setTimeout(() => answerQuestion(answer), 600);
    }
    return (
        <ListItem id={answer} className="Answer" divider onClick={handleClick}>{renderHTML()}</ListItem>
    );
}