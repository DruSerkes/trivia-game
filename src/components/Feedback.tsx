import * as React from 'react';



export const Feedback = () => (< p className="Feedback" data-testid="Feedback" />)