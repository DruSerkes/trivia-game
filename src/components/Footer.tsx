import React from "react";

export const Footer = () => {
  const year = new Date().getFullYear();
  return (
    <footer className="Footer">
      <p>
        <a
          data-testid="footer"
          href="https://github.com/druserkes"
          target="_blank"
          rel="noopener noreferrer"
        >
          Dru Serkes &copy;{year}
        </a>
      </p>
    </footer>
  );
};
