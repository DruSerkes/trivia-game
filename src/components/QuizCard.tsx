import * as React from 'react';
import { QuestionType } from '../redux/types';

type QuizCardProps = {
    question: QuestionType | null;
    answerQuestion: (answer: string) => void
}


export const QuizCard = ({ question, answerQuestion }: QuizCardProps) => {
    // Render the 3 incorrect answers and the correct answer
    // Do so with each being a separate Answer component
    // Answer component should accept props for question and isCorrect (optional bool)
    // When Answer is clicked, it should fire off answerQuestion with the answer passed in


    return (
        <div className="QuizCard">
            <h4>{question?.question}</h4>
            {/* RENDER 4 ANSWERS HERE */}
        </div>
    )
}