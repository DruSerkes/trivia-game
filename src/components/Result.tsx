import * as React from 'react';
import { Box, Typography } from '@material-ui/core'
import { renderColor } from '../helpers/renderColor'

type ResultTypes = {
    numCorrect: number;
    numAnswered: number;
}

export const Result = ({ numCorrect, numAnswered }: ResultTypes) => {
    return (
        <Box className="Result">
            <Typography variant="h3">Game Over!</Typography>
            <Typography variant="body1" style={{ color: renderColor(numCorrect) }}>You answered {numCorrect} out of {numAnswered} correct</Typography>
        </Box>
    )
}