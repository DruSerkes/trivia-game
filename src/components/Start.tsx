import * as React from 'react';
import { useState } from 'react';
import { Box, Button, TextField, FormControl, RadioGroup, FormControlLabel, Radio, FormLabel } from '@material-ui/core'

type StartProps = {
    startGame: (amount: string, difficulty: string, type: string) => void
}

export const Start = ({ startGame }: StartProps) => {
    const [amount, setAmount] = useState<string>('10');
    const [difficulty, setDifficulty] = useState<string>('');
    const [type, setType] = useState<string>('multiple');

    const handleChangeAmount = (e: React.ChangeEvent<HTMLInputElement>): void => setAmount(e.target.value);
    const handleChangeDifficulty = (e: React.ChangeEvent<HTMLInputElement>): void => setDifficulty(e.target.value);
    const handleChangeType = (e: React.ChangeEvent<HTMLInputElement>): void => setType(e.target.value);
    const handleSubmit = () => startGame(amount, difficulty, type);

    return (
        <Box className="Start">
            <TextField label="Number of Questions" type="number" value={amount} onChange={handleChangeAmount} />
            <FormControl component="fieldset">
                <FormLabel component="legend">Difficulty</FormLabel>
                <RadioGroup row aria-label="difficulty" name="difficulty" value={difficulty} onChange={handleChangeDifficulty}>
                    <FormControlLabel value="easy" control={<Radio />} label="Easy" labelPlacement="end" />
                    <FormControlLabel value="medium" control={<Radio />} label="Medium" labelPlacement="end" />
                    <FormControlLabel value="hard" control={<Radio />} label="Hard" labelPlacement="end" />
                    <FormControlLabel value="" control={<Radio />} label="Surprise Me!" labelPlacement="end" />
                </RadioGroup>
            </FormControl>
            <FormControl component="fieldset">
                <FormLabel component="legend">Type</FormLabel>
                <RadioGroup row aria-label="type" name="type" value={type} onChange={handleChangeType}>
                    <FormControlLabel value="multiple" control={<Radio />} label="Multiple Choice" labelPlacement="end" />
                    <FormControlLabel value="boolean" control={<Radio />} label="True/False" labelPlacement="end" />
                </RadioGroup>
            </FormControl>
            <Button variant="contained" fullWidth color="primary" onClick={handleSubmit}>Start New Game</Button>
        </Box >
    )
}