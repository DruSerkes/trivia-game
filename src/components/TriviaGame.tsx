/*
    * Redux lives here 
    * Local state to keep track of current question (which gets passed down) 
    * Local state to keep track of answers (correct, answered)  
    * Renders Start if no questions in state
    * Renders QuizCard otherwise
        * pass question data down 
    * Renders Result if â€¦
    * 
    * 
    * res.data.results[0-9]
        * set amount=10
        * set type=multiple
        * .question 
        * .correct_answer 
        * .incorrect_answers[0-2]
*/
import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { StoreType } from '../redux/store';
import { QuestionType } from '../redux/types';

// TODO
// 1. Create QuizCard component
// 2. Create Result component
// 3. import each here

const TriviaGame: React.FC = () => {
	const questions = useSelector((state: StoreType) => state.questions);
	const [ currentQuestion, setCurrentQuestion ] = useState<QuestionType | null>(null);
	const [ numAnswered, setNumAnswered ] = useState(0);
	const [ numCorrect, setNumCorrect ] = useState(0);

	const answerQuestion = (answer: string) => {
		if (currentQuestion && answer === currentQuestion.correct_answer) setNumCorrect((correct) => correct++);
		setNumAnswered((answered) => answered++);
		setCurrentQuestion(() => questions[numAnswered]);
	};

	return (
		<div className="TriviaGame">
			{numAnswered < 10 ? (
				<QuizCard question={currentQuestion} answerQuestion={answerQuestion} />
			) : (
				<Result numCorrect={numCorrect} />
			)}
		</div>
	);
};

export default TriviaGame;
