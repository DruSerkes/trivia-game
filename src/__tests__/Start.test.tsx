import React from 'react';
import { render } from '@testing-library/react';
import { Start } from '../components/Start'


test('Start Matches Snapshot ', () => {
    const { asFragment } = render(
        <Start startGame={() => { }} />
    );
    expect(asFragment()).toMatchSnapshot();
});