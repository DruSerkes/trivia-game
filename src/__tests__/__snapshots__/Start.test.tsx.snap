// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start Matches Snapshot  1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-1 Start"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Start New Game
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
